src/tracing/tracing.c

```diff
@@ -104,13 +104,24 @@ static char* xdebug_start_trace(char* fname, zend_string *script_filename, long
 	}
 
 	XG_TRACE(trace_handler) = xdebug_select_trace_handler(options);
-	XG_TRACE(trace_context) = (void*) XG_TRACE(trace_handler)->init(fname, script_filename, options);
+	if (XG_TRACE(trace_handler)->init) {
+		XG_TRACE(trace_context) = (void*) XG_TRACE(trace_handler)->init(fname, script_filename, options);
+	} else {
+		/* @todo emit and error to expose broken handler */
```

Comment:
I have no idea on how to properly emit a an error from xdebug here, but I think that, in order to avoid segfault, it should be done. As being a developer that just implemented one tracing handler, it would have helper me a lot to see an explicit error while tests were running.

Reply:
You can do something like:

```
xdebug_log_ex(XLOG_CHAN_TRACE, XLOG_CRIT, "HNDLR", "Broken trace handler for format '%d', missing 'init' handler", XINI_TRACE(trace_format));
```

Depending on the location, this might also automatically provide a link to the documentation, such as ``https://xdebug.org/docs/errors#CFG-C-REMOVED``